pkg/controller/deactivation/deactivation_controller_test.go

```diff
@@ -151,43 +151,123 @@ func TestReconcile(t *testing.T) {
+
+				// deactivating state should still be true
+				require.True(t, states.Deactivating(userSignupFoobar))
+
+				// deactivated state should still be false
+				require.False(t, userSignupFoobar.Spec.Deactivated)
+
+				t.Run("usersignup requeued after deactivating notification created for user", func(t *testing.T) {
+					// Set the notification status condition as sent
+					userSignupFoobar.Status.Conditions = []toolchainv1alpha1.Condition{
+						toolchainv1alpha1.Condition{
+							Type:               toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+							Status:             v1.ConditionTrue,
+							LastTransitionTime: metav1.Time{Time: time.Now()},
+							Reason:             toolchainv1alpha1.UserSignupDeactivatingNotificationCRCreatedReason,
+						},
+					}
+
+					r, req, cl := prepareReconcile(t, mur.Name, basicTier, mur, userSignupFoobar, config)
+
+					// when
+					res, err := r.Reconcile(req)
+					// then
+					require.NoError(t, err)
+					require.False(t, res.Requeue)
+					// The RequeueAfter should be ~about 3 days... let's accept if it's within 1 hour of that
+					require.WithinDuration(t, time.Now().Add(time.Duration(72)*time.Hour), time.Now().Add(res.RequeueAfter), time.Duration(1)*time.Hour)
+
+					// Reload the userSignup
+					require.NoError(t, cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar))
+
+					// deactivating state should still be true
+					require.True(t, states.Deactivating(userSignupFoobar))
+
+					// deactivated state should still be false
+					require.False(t, userSignupFoobar.Spec.Deactivated)
+
+					t.Run("usersignup should be deactivated", func(t *testing.T) {
+						// Set the notification status condition as sent, but this time 3 days in the past
+						userSignupFoobar.Status.Conditions = []toolchainv1alpha1.Condition{
+							toolchainv1alpha1.Condition{
+								Type:               toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+								Status:             v1.ConditionTrue,
+								LastTransitionTime: metav1.Time{Time: time.Now().Add(time.Duration(-3) * time.Hour * 24)},
+								Reason:             toolchainv1alpha1.UserSignupDeactivatingNotificationCRCreatedReason,
+							},
+						}
+
+						r, req, cl := prepareReconcile(t, mur.Name, basicTier, mur, userSignupFoobar, config)
+
+						// when
+						res, err := r.Reconcile(req)
+						// then
+						require.NoError(t, err)
+						require.False(t, res.Requeue)
+
+						// Reload the userSignup
+						require.NoError(t, cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar))
+
+						// deactivating state should still be true
+						require.True(t, states.Deactivating(userSignupFoobar))
+
+						// deactivated state should now be true also
+						require.True(t, userSignupFoobar.Spec.Deactivated)
+
```

Comment:
We can also check that the deactivation notification is created and the corresponding condition is true, right?

Reply:
As above.